flow/util/misc.py

```diff
@@ -359,37 +362,59 @@ def _get_parallel_executor(parallelization="none"):
 
     """
     if parallelization == "thread":
```

Comment:
This would be cleaner as a nested if with `progress` first then `parallelization`.